packages/react-ui-components/src/components/CodeEditor/CodeEditor.tsx

```diff
@@ -1,53 +1,45 @@
-import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import React, { useCallback, useEffect, useMemo } from 'react';
 import Editor, { useMonaco } from '@monaco-editor/react';
-import { DEFAULT_EDITOR_LANGUAGE, DEFAULT_EDITOR_THEME, DEFAULT_PADDING_TOP, FONT_NOT_FOUND_ERROR } from './config';
+import { DEFAULT_EDITOR_LANGUAGE, DEFAULT_EDITOR_OPTIONS, DEFAULT_EDITOR_THEME, KELVIN_CODE_EDITOR_THEME } from './config';
 import { ECodeEditorTheme, ICodeEditorProps, OnEditorChangeCallback } from './types';
 import { KvLoader } from '../stencil-generated';
 import { getEditorOptions } from './utils';
+import { useFontsApi } from '../../utils/useFontsApi';
+
+const CodeEditorLoader = () => <KvLoader isLoading />;
 
 export const KvCodeEditor = ({
 	code,
-	readOnly = false,
-	loadingComponent = <KvLoader isLoading />,
+	LoadingComponent = CodeEditorLoader,
 	language = DEFAULT_EDITOR_LANGUAGE,
 	theme = DEFAULT_EDITOR_THEME,
-	customTheme,
-	paddingTop = DEFAULT_PADDING_TOP,
+	customTheme = KELVIN_CODE_EDITOR_THEME,
+	customOptions = DEFAULT_EDITOR_OPTIONS,
 	onChange
 }: ICodeEditorProps) => {
 	const monaco = useMonaco();
-	const [isFontLoaded, setFontLoaded] = useState(false);
-
-	const options = useMemo(() => getEditorOptions({ readOnly, padding: { top: paddingTop } }), [readOnly, paddingTop]);
+	const options = useMemo(() => getEditorOptions(customOptions), [customOptions]);
+	const { checkFont, loadFont } = useFontsApi(options.fontFamily, options.fontSize);
 
-	const loadFont = useCallback(async () => {
-		try {
-			const { fontSize, fontFamily } = options;
-			await document.fonts.load(`${fontSize}px ${fontFamily}`);
-			setFontLoaded(true);
-		} catch (error) {
-			throw new Error(FONT_NOT_FOUND_ERROR);
-		}
-	}, [options.fontSize, options.fontFamily]);
+	const hasLoaded = useMemo(() => monaco && checkFont(), [monaco, checkFont]);
 
-	const setupEditor = useCallback(() => {
+	const defineCustomTheme = useCallback(() => {
 		if (!monaco || !customTheme) return;
 		monaco.editor.defineTheme(ECodeEditorTheme.Custom, customTheme);
-		monaco.editor.setTheme(ECodeEditorTheme.Custom);
-	}, [monaco, customTheme]);
+	}, [customTheme, monaco]);
 
 	const onTextChange: OnEditorChangeCallback = useCallback(value => onChange?.(value), [onChange]);
 
 	useEffect(() => {
 		loadFont();
-		setupEditor();
-	}, [loadFont, setupEditor]);
+		defineCustomTheme();
+	}, [loadFont, defineCustomTheme]);
```

Comment:
probably we need to add monaco to the dependency list, to reload the data if the instance change.

Reply:
`monaco` is already on the dependency list of `defineCustomTheme` and `hasLoaded`, should we include it here?